aptos-move/aptos-gas-profiling/src/profiler.rs

```diff
@@ -466,6 +469,28 @@ where
 
         res
     }
+
+    fn charge_dependency(
+        &mut self,
+        is_new: bool,
+        addr: &AccountAddress,
+        name: &IdentStr,
+        size: NumBytes,
+    ) -> PartialVMResult<()> {
+        let (cost, res) =
+            self.delegate_charge(|base| base.charge_dependency(is_new, addr, name, size));
+
+        if !cost.is_zero() {
```

Comment:
what's the meaning of zero here?

Reply:
This if statements makes sure that a dependency is only recorded if the cost is non-zero.

I might move the filtering of zero-cost items to a different place in the future. This is the gas profiler (an offline tool) so we should be able to change this easily.